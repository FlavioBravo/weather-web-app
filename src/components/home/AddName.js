import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddName = ({ setCities }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            // setCities(cities => [inputValue, ...cities]);
            setCities(cities => [inputValue]);
            setInputValue('');
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <h4>Filters</h4>
            <label>Name:</label>
            <p> {inputValue} </p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddName.propTypes = {
    setCities: PropTypes.func.isRequired
}