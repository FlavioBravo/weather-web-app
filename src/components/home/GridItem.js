import React from 'react';
import PropTypes from 'prop-types';
import { useHistory } from "react-router-dom";

export const GridItem = ({ id, name, state, country, coord }) => {

    const history = useHistory();

    const handleForecast = () => {
        history.push(`/forecast/${id}`);
    };

    return (
        <tr className="animate__animated animate__fadeIn" onClick={ handleForecast } >
            <td> { id } </td>
            <td> { name } </td>
            <td> { state } </td>
            <td> { country } </td>
            <td> { coord.lon } </td>
            <td> { coord.lat } </td>
        </tr>
    )
}

GridItem.propTypes = {
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    state: PropTypes.string.isRequired,
    country: PropTypes.string.isRequired,
    coord: PropTypes.object.isRequired,
}