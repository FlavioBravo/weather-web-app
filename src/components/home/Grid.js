import React from 'react';
import PropTypes from 'prop-types';

import { useFetchCities } from '../../hooks/useFetchCities';
import { GridItem } from './GridItem';

export const Grid = ({ city }) => {

    const { data: images, loading } = useFetchCities(city);

    return (
        <>
            <br />
            <h3 className="animate__animated animate__fadeIn"> {city} </h3>
            <br />
            { loading && <p className="animate__animated animate__flash">Loading</p>}

            <table className="">
                <thead>
                    <tr>
                        <th>ID</th>
                        <th>Name</th>
                        <th>State</th>
                        <th>Country</th>
                        <th>Longitude</th>
                        <th>Latitude</th>
                    </tr>
                </thead>
                <tbody>
                    {
                        images.map(img => (
                            <GridItem
                                key={img.id}
                                {...img}
                            />
                        ))
                    }
                </tbody>
            </table>
        </>
    )
}

Grid.propTypes = {
    city: PropTypes.string.isRequired
}