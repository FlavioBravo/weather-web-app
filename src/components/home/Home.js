import React, { useState } from 'react';

import { AddName } from './AddName';
import { Grid } from './Grid';

export const Home = ({ defaultCities = [] }) => {

    const [cities, setCities] = useState( defaultCities );

    return (
      <>
          <h2>Weather Web App</h2>
          <AddName setCities={ setCities } />
          <hr />

          <ol>
              {
                  cities.map( city  => (
                      <Grid 
                          key={ city }
                          city={ city }
                      />
                  ))
              }
          </ol>

      </>
  )
}
