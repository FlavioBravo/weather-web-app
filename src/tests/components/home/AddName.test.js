import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { AddName } from '../../../components/home/AddName';


describe('Pruebas en <AddName />', () => {

    const setCities = jest.fn();
    let wrapper = shallow( <AddName setCities={ setCities } /> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddName setCities={ setCities } /> );
    });

    
    test('debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe de cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate('change', { target: { value } });

        expect( wrapper.find('p').text().trim() ).toBe( value );
        
    })

    test('NO debe de postear la informaciÃ³n con submit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCities ).not.toHaveBeenCalled();

    })
    
    test('debe de llamar el setCategories y limpiar la caja de texto', () => {

        const value = 'Hola Mundo';

        // 1. simular el inputChange
        wrapper.find('input').simulate('change', { target: { value } });

        // 2. simular el submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        // 3. setCategories se debe de haber llamado
        expect( setCities ).toHaveBeenCalled();
        expect( setCities ).toHaveBeenCalledTimes(1);
        expect( setCities ).toHaveBeenCalledWith( expect.any(Function)  );

        // 4. el valor del input debe de estar ''
        expect( wrapper.find('input').prop('value') ).toBe('');

        
    })
    
    

})