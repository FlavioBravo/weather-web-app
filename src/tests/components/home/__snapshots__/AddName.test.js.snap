// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddName /> debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <h4>
    Filters
  </h4>
  <br />
  <label>
    Name:
  </label>
  <br />
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    placeholder="Write city name here ..."
    type="text"
    value=""
  />
</form>
`;
